### DISCUSSION TITLE
Gmail :Google no longer supports the use of third party apps . . .

### DISCUSSION THREAD

**Discussion by jcg31** [PHPMailer/PHPMailer#2694] [General]
So as of May 30, Google rolled out security precautions towards any apps not meeting their new security standards ( https://support.google.com/accounts/answer/6010255?hl=en#:~:text=To%20help%20keep%20your%20account,or%20Google%20Cloud%20Identity%20customers. ).

I have used imap to receive and PHPmailer to send emails for years without issue. Yesterday everything stopped working because of the Google change above. I went through the suggested process of getting 2-Step Verification set up and then receiving the app password Google provide and replacing the various instances of the current passwords with the App Password. I did that for both the imap and PHPmailer. The imap apps work fine now but changing just the password in the PHPmailer instances did not.

Has anyone come up with a solution for addressing the changing Google standard as it applies to PHPmailer?

**Comment by Synchro** [comment]
PHPMailer added XOAUTH2 support 5 years ago when Google first warned this would happen, and if you use that you won't have this problem, though of course it's OAuth so expect it to be confusing and complicated! The recent change is covered in the troubleshooting guide, but not in great detail. I don't know of any reason why the app password approach would not work for you – I don't use gmail often, but this has worked every time I've tried it. Can you describe exactly *how* it's not working for you?

**Comment by jcg31** [comment]
I am getting the following error message:
 SERVER -> CLIENT: 535-5.7.8 Username and Password not accepted. Learn more at 535 5.7.8 https://support.google.com/mail/?p=BadCredentials z24-20020a05660200d800b006694cd8805asm3849974ioe.53 - gsmtp 2022-06-08 15:47:09 SMTP ERROR: Password command failed

**Reply by Synchro** [reply]
There have been several other discussions about this same issue, such as in #2693, so I'll responding to things there since it as a good code example.

**Reply by jcg31** [reply]
Thanks!
